fl4health/clients/basic_client.py

```diff
@@ -47,13 +50,20 @@ def __init__(
                 each batch. Defaults to LossMeterType.AVERAGE.
             checkpointer (Optional[TorchCheckpointer], optional): Checkpointer to be used for client-side
                 checkpointing. Defaults to None.
+            metrics_reporter (Optional[MetricsReporter], optional): A metrics reporter instance to record the metrics
+                during the execution. Defaults to an instance of MetricsReporter with default init parameters.
         """
 
         self.data_path = data_path
         self.device = device
         self.metrics = metrics
         self.checkpointer = checkpointer
 
+        if metrics_reporter is not None:
+            self.metrics_reporter = metrics_reporter
+        else:
+            self.metrics_reporter = MetricsReporter()
```

Comment:
Each client generates a hash as its `client_name` (see line 66 just below). Do you think we should use that as the `run_id` for the metrics reporter if one isn't provided?

Reply:
Yes, that's a good idea. Will do.